server/indexers/transfers.ts

```diff
@@ -0,0 +1,103 @@
+import { PoolClient } from 'pg';
+import {
+  getTransferLastQueriedBlock,
+  setTransferLastQueriedBlock,
+  insertWallet,
+  insertNounOwnerInAuction,
+} from './queries';
+import { TransferEvent } from '../../typechain/NounsToken';
+
+type NounsTokenEventLog = TransferEvent.Log;
+
+const log = logger.child({ indexer: 'transfer' });
+
+export default async function transfers(
+  nounAddress: string,
+  connection: PoolClient,
+  provider: ethers.Provider,
+) {
+  log.info('Starting');
+  const nouns = NounsToken__factory.connect(nounAddress, provider);
+
+  async function maybeProcessEvent(event: NounsTokenEventLog) {
+    try {
+      log.debug('Processing event %s', event.eventName, { event });
+      processEvent(connection, event);
+    } catch (error) {
+      log.error(error);
+    }
+  }
+
+  nouns.on(nouns.filters.Transfer(), (...args) => {
+    // Ethers: the first N arguments are event args verbatim
+    // The event object we care about is the last argument
+    const payload = args[args.length - 1] as unknown as ContractEventPayload;
+
+    maybeProcessEvent(payload.log as unknown as NounsTokenEventLog);
+  });
+
+  const result = await getTransferLastQueriedBlock.run(undefined, connection);
+
+  const lastQueriedBlock = result[0]?.value || 12985450; // Nouns Deployment -1
+  let lastBlockNumber = lastQueriedBlock;
+
+  log.debug('Loaded state', { lastQueriedBlock });
+  const currentBlockNumber = await provider.getBlockNumber();
+  const limit = 1_000_000;
+  const pages = Math.ceil((currentBlockNumber - lastQueriedBlock) / limit);
+
+  for (let page = 0; page <= pages; page += 1) {
+    const blockFrom = lastQueriedBlock + page * limit + 1;
+    const blockTo = Math.min(currentBlockNumber, lastQueriedBlock + (page + 1) * limit);
+    if (blockFrom > currentBlockNumber) {
+      break;
+    }
+
+    log.debug('Querying events', { blockFrom, blockTo });
+    const events: Array<NounsTokenEventLog> = await nouns.queryFilter(
+      nouns.filters.Transfer(),
+      blockFrom,
+      blockTo,
+    );
+
+    for (const event of events) {
+      await maybeProcessEvent(event);
+      lastBlockNumber = Math.max(lastBlockNumber, event.blockNumber);
+    }
+
+    await setTransferLastQueriedBlock.run({ lastBlockNumber }, connection);
+  }
+
+  // Wait forever, otherwise the connection is released and the event handlers
+  // won't be able to run SQL
+  await new Promise((resolve) => {});
+}
+
+async function processEvent(connection: PoolClient, eventLog: NounsTokenEventLog) {
+  if (eventLog.eventName === 'Transfer') {
+    const { to, tokenId } = (eventLog as TransferEvent.Log).args;
+
+    if (Number(tokenId) % 10 === 0) return; // Remove nouns that are not associated with auctions
+
+    await insertNounOwnerInAuction.run(
+      {
+        id: Number(tokenId),
+        owner: to,
+      },
+      connection,
+    );
+    await insertWallet.run(
+      {
+        address: to,
+      },
+      connection,
+    );
```

Comment:
I think instead of inserting a wallet here, we can make the `findUnindexedWallets` query also look for `noun.owner` addresses that are missing